core/GlobalState.h

```diff
@@ -186,6 +186,7 @@ class NameHash {
 class FileTable {
     std::vector<std::shared_ptr<File>> files;
     UnorderedMap<std::string, FileRef> fileRefByPath;
+    std::vector<UnorderedSet<core::SymbolRef>> symbolsReferencedByFile;
```

Comment:
This is particularly dangerous. We currently have zero places where the FileTable depends on Symbol information. Files as objects are meant to live independent of any symbol table: they're very frequently handled as `shared_ptr<File>` objects that are moved from one GlobalState to another, which might have different symbols and thus different `SymbolRef` IDs.

This in particular is going to require some careful attention. I would also love @elliottt's eyes on this, because I know he has a lot of valuable context about the invariants we need to maintain on the file table at this point.

Reply:
I'm not suggesting that this move into the PackageDB, btw. I'm saying, maybe it should be outside the file table (in my mind, one day the PackageDB and GlobalState will be the same, because the list of packages will move into GlobalState's symbol table, but for example as a part of that `packageForFile_` is not going to move into the file table, I don't think)